src/main/java/com/stripe/net/ApiRequest.java

```diff
@@ -0,0 +1,33 @@
+package com.stripe.net;
+
+import java.util.List;
+import java.util.Map;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+public class ApiRequest {
+  private BaseAddress baseAddress;
+  private ApiResource.RequestMethod method;
+  private String path;
+  private Map<String, Object> params;
+  private RequestOptions options;
+  private ApiMode apiMode;
+
+  @Setter private List<String> usage;
+
+  public ApiRequest(
+      BaseAddress baseAddress,
+      ApiResource.RequestMethod method,
+      String path,
+      Map<String, Object> params,
```

Comment:
Taking into account https://github.com/stripe/stripe-java/pull/1717, it doesn't feel right that the base type has `private Map<String, Object> params` while RawRequest adds a `
  @Getter private String rawContent;` as well. We get a type with two things that can become a body.

Should we define a base type without params and then have a `Map` and Raw subtypes?

Reply:
Works for me!

Reply:
Done in [BaseApiRequest](https://github.com/stripe/stripe-java/pull/1702/commits/8c03e55ed2987214ec34066f8c13d83c7eea26f2)